internal/configs/transportserver_test.go

```diff
@@ -366,3 +365,89 @@ func TestGenerateUnixSocket(t *testing.T) {
 		t.Errorf("generateUnixSocket() returned %q but expected %q", result, expected)
 	}
 }
+
+func TestGenerateTransportServerHealthChecks(t *testing.T) {
+	tests := []struct {
+		healthCheck *conf_v1alpha1.HealthCheck
+		expected    *version2.StreamHealthCheck
+	}{
+		{
+			healthCheck: &conf_v1alpha1.HealthCheck{
+				Enabled:  false,
+				Timeout:  "30s",
+				Jitter:   "30s",
+				Port:     80,
+				Interval: "20s",
+				Passes:   4,
+				Fails:    5,
+			},
+			expected: &version2.StreamHealthCheck{
+				Enabled:  false,
+				Timeout:  "5s",
+				Jitter:   "0",
+				Port:     0,
+				Interval: "5s",
+				Passes:   1,
+				Fails:    1,
+			},
+		},
+		{
+			healthCheck: &conf_v1alpha1.HealthCheck{},
+			expected: &version2.StreamHealthCheck{
+				Enabled:  false,
+				Timeout:  "5s",
+				Jitter:   "0",
+				Port:     0,
+				Interval: "5s",
+				Passes:   1,
+				Fails:    1,
+			},
+		},
+		{
+			healthCheck: &conf_v1alpha1.HealthCheck{
+				Enabled:  true,
+				Timeout:  "30s",
+				Jitter:   "30s",
+				Port:     80,
+				Interval: "20s",
+				Passes:   4,
+				Fails:    5,
+			},
+			expected: &version2.StreamHealthCheck{
+				Enabled:  true,
+				Timeout:  "30s",
+				Jitter:   "30s",
+				Port:     80,
+				Interval: "20s",
+				Passes:   4,
+				Fails:    5,
+			},
+		},
+		{
+			healthCheck: &conf_v1alpha1.HealthCheck{
+				Enabled: true,
+				Timeout: "30s",
+				Jitter:  "30s",
+				Port:    80,
+			},
+			expected: &version2.StreamHealthCheck{
+				Enabled:  true,
+				Timeout:  "30s",
+				Jitter:   "30s",
+				Port:     80,
+				Interval: "5s",
+				Passes:   1,
+				Fails:    1,
+			},
+		},
+	}
+
+	for _, test := range tests {
+		result := generateTransportServerHealthCheck(test.healthCheck)
+		if !reflect.DeepEqual(result, test.expected) {
+			fmt.Printf("Input : %+v\n", test.healthCheck)
```

Comment:
for comparing large structs, better to use this approach, as it prints a nice diff, so it's easy to spot the mismatch:
```
		if diff := cmp.Diff(test.expected, result); diff != "" {
			t.Errorf("generatePolicies() '%v' mismatch (-want +got):\n%s", test.msg, diff)
		}
```